mypy_django_plugin/transformers/fields.py

```diff
@@ -201,6 +205,156 @@ def drop_combinable(_type: MypyType) -> Optional[MypyType]:
     return helpers.reparametrize_instance(default_return_type, args)
 
 
+class M2MThrough(NamedTuple):
+    arg: Optional[Expression]
+    model: ProperType
+
+
+class M2MTo(NamedTuple):
+    arg: Expression
+    model: ProperType
+    self: bool  # ManyToManyField('self', ...)
+
+
+class M2MArguments(NamedTuple):
+    to: M2MTo
+    through: Optional[M2MThrough]
+
+
+def get_model_from_expression(
+    expr: Expression,
+    *,
+    api: Union[TypeChecker, SemanticAnalyzer],
+    django_context: DjangoContext,
+) -> Optional[ProperType]:
+    """
+    Attempts to resolve an expression to a 'TypeInfo' instance. Any lazy reference
+    argument(e.g. "<app_label>.<object_name>") to a Django model is also attempted.
+    """
+    # TODO: Handle settings.AUTH_USER_MODEL?
```

Comment:
To expand a bit more here; I have an idea that we could also attempt to resolve a `MemberExpr` pointing towards `settings.AUTH_USER_MODEL`. As such it's possible to pick up its lazy reference, something like

```python
if isinstance(expr, MemberExpr) and ...:
    label = django_context.settings.AUTH_USER_MODEL
    # Below here is same as when `isinstance(expr, NameExpr)`
    fullname = django_context.model_class_fullnames_by_label.get(label)
        if fullname is not None:
            model_info = helpers.lookup_fully_qualified_typeinfo(api, fullname)
            if model_info is not None:
                return Instance(model_info, [])
            elif isinstance(api, SemanticAnalyzer) and not api.final_iteration:
                # Getting this far, where Django matched the reference but we still can't
                # find it, we want to defer
                api.defer()
```

Like so, there could be another helper:

```python
def resolve_lazy_reference(expr: Expression) -> Optional[TypeInfo]:
    ...
```

Which focuses on converting any `StrExpr("<app_label>"."<object_name>")` to a `TypeInfo`. This could probably be reused in multiple other places e.g. `Apps.get_model(...)` and `to=` argument for `ForeignKey`, `OneToOneField`, etc.